File: Day08_Notification/app/src/main/java/day08/nguyenhongphat0/day08_notification/MainActivity.java
package day08.nguyenhongphat0.day08_notification;

import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.app.TaskStackBuilder;
import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.support.v4.app.NotificationCompat;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {
    private NotificationManager manager;
    private int notiID = 6789;
    private int numMsg = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void clickToSend(View view) {
        NotificationCompat.Builder builder = new NotificationCompat.Builder(this, "channel08");
        builder.setContentTitle("Content title");
        builder.setContentText("Content text: Message has been received");
        builder.setTicker("Ticker: Message Alert");
        builder.setSmallIcon(R.drawable.ic_action_unread);
        builder.setNumber(++numMsg);
        Intent intent = new Intent(this, NotificationDetailActivity.class);
        TaskStackBuilder stack = TaskStackBuilder.create(this);
        stack.addNextIntent(intent);
        PendingIntent pendingIntent = stack.getPendingIntent(0, PendingIntent.FLAG_UPDATE_CURRENT);
        builder.setContentIntent(pendingIntent);
        manager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
            NotificationChannel chanel = new NotificationChannel("channel08", "Testing notification channel", NotificationManager.IMPORTANCE_HIGH);
            manager.createNotificationChannel(chanel);
        }
        manager.notify(numMsg, builder.build());
    }

    public void clickToCancel(View view) {
        manager  = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
        manager.cancel(notiID);
    }

    public void clickToUpdate(View view) {
    }

    public void clickToSendMultipleTime(View view) {
        String channelID = "channel08";
        NotificationCompat.Builder builder = new NotificationCompat.Builder(this, channelID);
        builder.setContentTitle("Content title" + numMsg);
        builder.setContentText("Content text: Message has been received" + numMsg);
        builder.setTicker("Ticker: Message Alert");
        builder.setSmallIcon(R.drawable.ic_action_unread);
        builder.setNumber(++numMsg);
        NotificationCompat.InboxStyle inboxStyle = new NotificationCompat.InboxStyle();
        String[] events = {"HP+", "Mana+", "Luck+"};
        inboxStyle.setBigContentTitle("LV UP");
        for (int i = 0; i < events.length; i++) {
            inboxStyle.addLine(events[i]);
        }
        builder.setStyle(inboxStyle);
        Intent intent = new Intent(this, NotificationDetailActivity.class);
        TaskStackBuilder stack = TaskStackBuilder.create(this);
        stack.addNextIntent(intent);
        PendingIntent pendingIntent = stack.getPendingIntent(0, PendingIntent.FLAG_UPDATE_CURRENT);
        builder.setContentIntent(pendingIntent);
        manager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            manager.createNotificationChannel(new NotificationChannel(channelID, "Multiple Notify", NotificationManager.IMPORTANCE_DEFAULT));
        }
        manager.notify(notiID, builder.build());
    }
}

File: kotlin/Day08/app/src/main/java/day08/nguyenhongphat0/day08/MainActivity.kt
package day08.nguyenhongphat0.day08

import android.app.*
import android.content.Context
import android.content.Intent
import android.os.Build
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v4.app.NotificationCompat
import android.widget.Button
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        sendNotiButton.setOnClickListener {
            val channelId = "channel08"
            val intent = Intent(this, MainActivity::class.java)
            val stack = TaskStackBuilder.create(this)
            stack.addNextIntent(intent)
            val pendingIntent = stack.getPendingIntent(0, PendingIntent.FLAG_UPDATE_CURRENT)
            val notification = NotificationCompat.Builder(this, channelId)
                .setContentTitle(notiTitle.text)
                .setContentText("Lorem ipsum dolor sit amer")
                .setContentInfo("Extra information")
                .setSmallIcon(R.drawable.abc_ic_star_black_36dp)
                .setContentIntent(pendingIntent)
                .build()
            val manager = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                val channel = NotificationChannel(channelId, "Testing channel", NotificationManager.IMPORTANCE_HIGH)
                manager.createNotificationChannel(channel)
            }
            manager.notify(0, notification)
        }
    }
}
